core/src/test/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/ReplaceRootDslTest.kt

```diff
@@ -0,0 +1,349 @@
+
+                // when
+                val stage = replaceRoot {
+                    newRoot {
+                        nested(Test::a) {
+                            Test::b set Test::d
+                            Test::c set {
+                                add { of(1) and 2 }
+                            }
+                            Test::d setByField "c"
+                            nested("e") {
+                                "f" set 1
+                            }
+                        }
+                    }
+                }
+
+                // then
+                stage.shouldBeJson(
+                    """
+                {
+                  "${'$'}replaceRoot": {
+                    "newRoot": {
+                      "a.b": "${'$'}d",
+                      "a.c": {
+                        "${'$'}add": [
+                          1,
+                          2
+                        ]
+                      },
+                      "a.d": "${'$'}c",
+                      "a.e.f": 1
+                    }
+                  }
+                }
+                    """.trimIndent(),
+                )
+            }
+        }
+
+        "FieldReplacement" - {
+            "should set new root from a field" {
+                // given, when
+                val stage = replaceRoot {
+                    newRoot("a")
+                }
+
+                // then
+                stage.shouldBeJson(
+                    """
+                {
+                  "${'$'}replaceRoot": {
+                    "newRoot": "${'$'}a"
+                  }
+                }
+                    """.trimIndent(),
+                )
+            }
+
+            "should set new root from a field path" {
+                // given
+                data class Test(val a: Long)
+
+                // when
+                val stage = replaceRoot {
+                    newRoot(Test::a)
+                }
+
+                // then
+                stage.shouldBeJson(
+                    """
+                {
+                  "${'$'}replaceRoot": {
+                    "newRoot": "${'$'}a"
+                  }
+                }
+                    """.trimIndent(),
+                )
+            }
+        }
+    }
+
+    "AggregationExpressions" - {
+        "should set new root from aggregation expression" {
+            // given, when
+            val stage = replaceRoot {
+                newRoot<AggregationExpression> {
+                    add { of(1) and 2 }
+                }
+            }
+
+            // then
+            stage.shouldBeJson(
+                """
+                {
+                  "${'$'}replaceRoot": {
+                    "newRoot": {
+                      "${'$'}add": [
+                        1,
+                        2
+                      ]
+                    }
+                  }
+                }
+                """.trimIndent(),
+            )
+        }
+    }
```

Comment:
How about this?

Reply:
Sound great!
but we don't have to `newRoot` method to be reified, as we do not need type info in runtime.

```kt
    fun <T : AggregationExpression> newRoot(
        configuration: AggregationExpressionDsl.() -> T,
    ): ReplaceRootOperation =
        ReplaceRootOperation.builder().withValueOf(AggregationExpressionDsl().configuration())
```

Reply:
I understand what you mean. thank you 👍🏻